### ISSUE TITLE
Unable to synchronise preferences to account - "No authentication providers are available".

### ISSUE THREAD

**Issue by RokeJulianLockhart** [microsoft/vscode#232648]
<table><tr>
	<th>Does this issue occur when all extensions are disabled?</th>
	<td>Yes</td>
</tr></table>

#### **Environment**

<table><tr>
	<th>VS Code Version</th>
	<td>
		<ol>
			<li>
				<pre><code>rpm -qa 'code'</code></pre>
			</li>
			<li>
				<pre><code>code-1.95.0-1730153639.el8.x86_64</code></pre>
			</li>
		</ol>
	</td>
</tr><tr>
	<th>OS Version</th>
	<td><a href="https://download.fedoraproject.org/pub/fedora/linux/releases/40/Spins/x86_64/iso/Fedora-KDE-Live-x86_64-40-1.14.iso">https://download.fedoraproject.org/pub/fedora/linux/releases/40/Spins/x86_64/iso/Fedora-KDE-Live-x86_64-40-1.14.iso</a></td>
</tr></table>

I installed VS Code as https://github.com/microsoft/vscode-docs/blob/e0b395019dd4a064ed5445daebc639e108356b19/docs/setup/linux.md#rhel-fedora-and-centos-based-distributions instructs.

`kinfo` returns:

> ```YAML
> Operating System: Fedora Linux 40
> KDE Plasma Version: 6.2.2
> KDE Frameworks Version: 6.7.0
> Qt Version: 6.7.2
> Kernel Version: 6.11.5-200.fc40.x86_64 (64-bit)
> Graphics Platform: Wayland
> Processors: 12 × AMD Ryzen 5 7600X 6-Core Processor
> Memory: 30.4 GiB of RAM
> Graphics Processor: AMD Radeon RX 5700
> ```

#### **Steps to Reproduce**

1. Invoke `code`.
2. Invoke the "Manage" sidebar item's context menu.
3. Select the account synchronisation option.

**Comment by RokeJulianLockhart** [comment]
Per https://stackoverflow.com/revisions/71538540/1, https://github.com/microsoft/vscode/issues/144471#issuecomment-1059824360 remediated this:

<blockquote>

```log
PS /home/RokeJulianLockhart> sudo dnf install gnome-keyring libsecret libgnome-keyring                     
[sudo] password for RokeJulianLockhart: 
Last metadata expiration check: 1:50:07 ago on Wed 30 Oct 2024 14:57:45 GMT.
Package gnome-keyring-46.2-1.fc40.x86_64 is already installed.
Package libsecret-0.21.4-2.fc40.x86_64 is already installed.
Dependencies resolved.
============================================================================================================================================================================================================================================================================================================================
 Package                                                                           Architecture                                                            Version                                                                            Repository                                                               Size
============================================================================================================================================================================================================================================================================================================================
Installing:
 libgnome-keyring                                                                  x86_64                                                                  3.12.0-28.fc40                                                                     fedora                                                                  126 k

Transaction Summary
============================================================================================================================================================================================================================================================================================================================
Install  1 Package

Total download size: 126 k
Installed size: 306 k
Is this ok [y/N]: y
Downloading Packages:
libgnome-keyring-3.12.0-28.fc40.x86_64.rpm                                                                                                                                                                                                                                                  611 kB/s | 126 kB     00:00    
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                                                                                                                                       128 kB/s | 126 kB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                                                                                                                    1/1 
  Installing       : libgnome-keyring-3.12.0-28.fc40.x86_64                                                                                                                                                                                                                                                             1/1 
  Running scriptlet: libgnome-keyring-3.12.0-28.fc40.x86_64                                                                                                                                                                                                                                                             1/1 

Installed:
  libgnome-keyring-3.12.0-28.fc40.x86_64                                                                                                                                                                                                                                                                                    

Complete!
```

</blockquote>

**Comment by TylerLeonhardt** [comment]
It can't be a required package because different window managers will use different implementations of a keyring.

What's surprising to me is that you should have gotten a notification that included a link to these docs:
https://code.visualstudio.com/docs/editor/settings-sync#_troubleshooting-keychain-issues

So you're saying you saw no notification complaining about the keyring? Was there anything in the terminal that you launched VS Code in?

**Comment by RokeJulianLockhart** [comment]
> It can't be a required package because different window managers will use different implementations of a keyring.

@TylerLeonhardt, I'm using v6.2 of the Qt-based KDE Plasma DE, yet installing the aforementioned *GNOME* keyring appeared to remediate this problem (although I reinitialized Code too). Alongside the fact that, to my knowledge, Electron delegates its GUI solely to GTK on all Linux-based OSes, unlike Chromium itself, I estimate it may be necessary irrespective of the DE.

However, considering that Code had worked fine until this point, I don't know what to make of this.

> What's surprising to me is that you should have gotten a notification that included a link to these docs: https://code.visualstudio.com/docs/editor/settings-sync#_troubleshooting-keychain-issues

I certainly did not, which indeed is strange, since for every alternative failure I've experienced until now, a counterpart notification leading to similar documentation *did* appear. I doubt I could

> So you're saying you saw no notification complaining about the keyring? Was there anything in the terminal that you launched VS Code in?

The sole notification I saw - I verified approximately 3 times - was "No authentication providers are available".

Regarding the terminal, I appear to have inaccurately recorded my reproduction steps - I actually launched it from the `.desktop` file (as is standard) but recorded `code` to differentiate the installation from `code-insiders` before I had added the environment information to the report. Apologies.

...had I launched it from the terminal, I would indeed have had a log.

**Comment by TylerLeonhardt** [comment]
@RokeJulianLockhart I take it it would be a lot of work to get the repro back? If not, I would love to see the output of 
```
code --verbose --vmodule="*/components/os_crypt/*=1"
```
https://code.visualstudio.com/docs/editor/settings-sync#_troubleshooting-keychain-issues

I am curious to know if our logs say that that desktop environment was properly detected.
https://source.chromium.org/chromium/chromium/src/+/main:base/nix/xdg_util.cc;l=102-196


I just booted up a Fedora 40 VM with KDE and got what I expected:
![Image](https://github.com/user-attachments/assets/b35731d2-5d6c-4154-b4ed-9eef25e2ded1)

and since I didn't agree to it/did it fast enough I got:
![Image](https://github.com/user-attachments/assets/0c8836c8-723a-44cb-bd18-15aebc41b1c4)

which showed the troubleshooting guide.

I am actually now thinking in this instance you hit https://github.com/microsoft/vscode/issues/232516 ... as "No authentication providers are available" was a symptom of that issue that I had experienced.

**Comment by RokeJulianLockhart** [comment]
1. > I take it it would be a lot of work to get the repro back?

   @TylerLeonhardt, I tried uninstalling `libgnome-keyring` top ascertain whether it did actually affect anything:

   1. ```sh
      sudo dnf remove 'libgnome-keyring'
      ```

   1. <details>

      <blockquote>

      ```log         
      [sudo] password for RokeJulianLockhart: 
      Dependencies resolved.
      ============================================================================================================================================================================================================================================================================================================================
       Package                                                                           Architecture                                                            Version                                                                           Repository                                                                Size
      ============================================================================================================================================================================================================================================================================================================================
      Removing:
       libgnome-keyring                                                                  x86_64                                                                  3.12.0-28.fc40                                                                    @fedora                                                                  306 k

      Transaction Summary
      ============================================================================================================================================================================================================================================================================================================================
      Remove  1 Package

      Freed space: 306 k
      Is this ok [y/N]: y
      Running transaction check
      Transaction check succeeded.
      Running transaction test
      Transaction test succeeded.
      Running transaction
        Preparing        :                                                                                                                                                                                                                                                                                                    1/1 
        Erasing          : libgnome-keyring-3.12.0-28.fc40.x86_64                                                                                                                                                                                                                                                             1/1 

      Removed:
        libgnome-keyring-3.12.0-28.fc40.x86_64                                                                                                                                                                                                                                                                                    

      Complete!
      ```

   </blockquote>

   </details>

   ...and it did log me out, so I'll not be surprised if installing the keyring somehow remediated it. (Un)fortunately, though, that's all it did - I'm unable to recreate the problem.

1. > I am actually now thinking in this instance you hit https://github.com/microsoft/vscode/issues/232516 ... as "No authentication providers are available" was a symptom of that issue that I had experienced.

   Thanks for that! That sounds exactly like this issue, since it occurred after an update, and I'm currently on that version too, according to `rpm -qa` (although I did *just* do another update, so it might have been the version just before).

**Comment by TylerLeonhardt** [comment]
Lets then call this as a duplicate of https://github.com/microsoft/vscode/issues/232516 for now, especially since you're unblocked, and we'll focus on that issue. If after that is fixed, and you still see the issue, we can take a deeper look.

**Comment by RokeJulianLockhart** [comment]
@TylerLeonhardt, this is very strange. I just did a system update - see https://discussion.fedoraproject.org/t/how-to-upgrade-to-a-different-release-via-the-shell/135070/4 - and upon invoking VS Code, saw that it had disconnected itself again.

Immediately, I exited it and invoked `code --verbose --vmodule="*/components/os_crypt/*=1"`, as directed. However, on that launch, it was connected again.

I've uploaded the log file at https://gist.github.com/RokeJulianLockhart/ae7aa0cfc9dce657c566706d868be88f#file-20241101t155549-0000-txt-log, but I imagine it's not of much use?
